src/v/storage/api.cc

```diff
@@ -52,4 +63,61 @@ void api::handle_disk_notification(
     }
 }
 
+ss::future<> api::monitor() {
+    vassert(ss::this_shard_id() == 0, "Run on wrong core");
+
+    /*
+     * The control loop should run periodically, but the current setting of 2
+     * seconds which provides decent responsiveness is too low to be used in
+     * practice long term. A follow up change will increase this setting and
+     * then allow cores to wake up this thread on demand to provide reactivity
+     * without frequent polling.
+     */
+    constexpr auto frequency = std::chrono::seconds(2);
+
+    while (true) {
```

Comment:
Perhaps while !_gate.is_closed()?   Even if we rely on broken semaphores in practice, it makes it a more readable "run til shutdown"